cel/pkg/reconciler/cel/celrun.go

```diff
@@ -0,0 +1,162 @@
+func (r *Reconciler) ReconcileKind(ctx context.Context, run *v1alpha1.Run) reconciler.Event {
+	logger := logging.FromContext(ctx)
+	logger.Infof("Reconciling Run %s/%s", run.Namespace, run.Name)
+
+	// If the Run has not started, initialize the Condition and set the start time.
+	if !run.HasStarted() {
+		logger.Infof("Starting new Run %s/%s", run.Namespace, run.Name)
+		run.Status.InitializeConditions()
+		// In case node time was not synchronized, when controller has been scheduled to other nodes.
+		if run.Status.StartTime.Sub(run.CreationTimestamp.Time) < 0 {
+			logger.Warnf("Run %s/%s createTimestamp %s is after the Run started %s", run.Namespace, run.Name, run.CreationTimestamp, run.Status.StartTime)
+			run.Status.StartTime = &run.CreationTimestamp
+		}
+	}
+
+	if run.IsDone() {
+		logger.Infof("Run %s/%s is done", run.Namespace, run.Name)
+		return nil
+	}
+
+	if err := validate(run); err != nil {
+		logger.Errorf("Run %s/%s is invalid because of %s", run.Namespace, run.Name, err)
+		run.Status.MarkRunFailed(ReasonFailedValidation,
+			"Run can't be run because it has an invalid spec - %v", err)
+		return nil
+	}
+
+	// Create a program environment configured with the standard library of CEL functions and macros
+	env, err := cel.NewEnv(cel.Declarations())
+	if err != nil {
+		logger.Errorf("Couldn't create a program env with standard library of CEL functions & macros when reconciling Run %s/%s: %v", run.Namespace, run.Name, err)
+		return err
+	}
+
+	var runResults []v1alpha1.RunResult
+	for _, param := range run.Spec.Params {
+		// Combine the Parse and Check phases CEL program compilation to produce an Ast and associated issues
+		ast, iss := env.Compile(param.Value.StringVal)
+		if iss.Err() != nil {
+			logger.Errorf("CEL expression %s could not be parsed when reconciling Run %s/%s: %v", param.Name, run.Namespace, run.Name, iss.Err())
+			run.Status.MarkRunFailed(ReasonSyntaxError,
+				"CEL expression %s could not be parsed", param.Name, iss.Err())
+			return nil
+		}
+
+		// Generate an evaluable instance of the Ast within the environment
+		prg, err := env.Program(ast)
+		if err != nil {
+			logger.Errorf("CEL expression %s could not be evaluated when reconciling Run %s/%s: %v", param.Name, run.Namespace, run.Name, err)
+			run.Status.MarkRunFailed(ReasonEvaluationError,
+				"CEL expression %s could not be evaluated", param.Name, err)
+			return nil
+		}
+
+		// Evaluate the CEL expression (Ast)
+		out, _, err := prg.Eval(map[string]interface{}{})
+		if err != nil {
+			logger.Errorf("CEL expression %s could not be evaluated when reconciling Run %s/%s: %v", param.Name, run.Namespace, run.Name, err)
+			run.Status.MarkRunFailed(ReasonEvaluationError,
+				"CEL expression %s could not be evaluated", param.Name, err)
+			return nil
+		}
+
+		// Evaluation of CEL expression was successful
+		logger.Infof("CEL expression %s evaluated successfully when reconciling Run %s/%s", param.Name, run.Namespace, run.Name)
```

Comment:
OTOH, that value might be large/spammy, and it might even be something sensitive like a secret (though users shouldn't pass secrets via results/params, but 🤷‍♂️ ). I'd like to bias toward not logging this result out-of-the-box.